vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -1155,26 +1155,53 @@ void LedgerImpl::DeleteActivityInfo(
   bat_database_->DeleteActivityInfo(publisher_key, callback);
 }
 
-void LedgerImpl::ClearServerPublisherList(ledger::ResultCallback callback) {
-  bat_database_->ClearServerPublisherList(callback);
+bool LedgerImpl::ShouldFetchServerPublisherInfo(base::Time last_updated_time) {
+  return bat_publisher_->ShouldFetchServerPublisherInfo(last_updated_time);
 }
 
-void LedgerImpl::InsertServerPublisherList(
-    const std::vector<ledger::ServerPublisherPartial>& list,
+void LedgerImpl::SearchPublisherList(
+    const std::string& publisher_key,
+    ledger::SearchPublisherListCallback callback) {
+  bat_database_->SearchPublisherList(publisher_key, callback);
+}
+
+void LedgerImpl::ResetPublisherList(
+    std::unique_ptr<braveledger_publisher::PublisherListReader> reader,
     ledger::ResultCallback callback) {
-  bat_database_->InsertServerPublisherList(list, callback);
+  bat_database_->ResetPublisherList(std::move(reader), callback);
 }
 
-void LedgerImpl::InsertPublisherBannerList(
-    const std::vector<ledger::PublisherBanner>& list,
+void LedgerImpl::InsertServerPublisherInfo(
+    const ledger::ServerPublisherInfo& server_info,
     ledger::ResultCallback callback) {
-  bat_database_->InsertPublisherBannerList(list, callback);
+  bat_database_->InsertServerPublisherInfo(server_info, callback);
 }
 
 void LedgerImpl::GetServerPublisherInfo(
     const std::string& publisher_key,
     ledger::GetServerPublisherInfoCallback callback) {
-  bat_database_->GetServerPublisherInfo(publisher_key, callback);
+  bat_database_->GetServerPublisherInfo(
+      publisher_key,
+      std::bind(&LedgerImpl::OnServerPublisherInfoLoaded,
+          this, _1, publisher_key, callback));
+}
+
+void LedgerImpl::OnServerPublisherInfoLoaded(
+    ledger::ServerPublisherInfoPtr server_info,
+    const std::string& publisher_key,
+    ledger::GetServerPublisherInfoCallback callback) {
+  if (bat_publisher_->ShouldFetchServerPublisherInfo(server_info.get())) {
```

Comment:
After getting the data from the database layer, we check if the record is expired, and if so, re-fetch it from the private CDN.